geps/gep-1619.md

```diff
@@ -62,14 +62,110 @@ Header-based stateful sessions are achieved by a backend or gateway providing an
+also plays in a role in session initiation. If a client doesn't want a request to belong an existing session, it can
+remove the session cookie or identifier to signal the gateway or backend to recreate the session.
 
+Let's focus on the two primary sources: gateways and the backend. The following rules apply based on who initiates the session:
+- If the gateway initiates the session, the backend will be presented with session attributes regardless if it enabled them.
 - If the backend initiates the session, the gateway should allow this and not force persistence connections, unless specifically configured to. The gateway may decode and alter the cookie established by the backend to achieve session persistence.
-- If the gateway initiates the session, the backend will be presented with session attributes regardless if it enabled them. 
 
-The client application also plays in a role in session initiation. If a client doesn't want a request to belong an existing session, it can remove the session cookie or identifier to signal the gateway or backend to recreate the session.
+In the next sections, we will take a closer look at the initiation of sessions in both the gateway and the backend.
+Please note that in the next sections, we are examining the various scenarios in which a session can be initiated. We
+are not prescribing specific implementations for session persistence. The intention is to understand the possibilities
+and behaviors related to session initiation while the [API](#api) section will provide more details on specific
+implementation details.
+
+#### Gateway Initiated Sessions
+
+This represents the most straightforward scenario for utilizing cookies. When a request is made, the gateway includes
+the `set-cookie` header in the final response, prompting the client to generate a cookie. This cookie is subsequently
+used in future requests, allowing the gateway to consistently choose the same upstream, establishing a persistent
+session.
+
+Here an example implementation of this process:
+```mermaid
+sequenceDiagram
+    actor C as Client
+    participant G as Gateway
+    participant B as Backend
+    C->>+G: Request Web Page
+    activate G
+    G->>+B: Request
+    B-->>-G: Response
+    G->>G: Add set-cookie header
+    G-->>-C: Response<br>[set-cookie]
+    Note right of G: [set-cookie] indicates a response<br> with a set-cookie header
+    C->>C: Create Cookie<br>from set-cookie header
+    Note right of C: [cookie] indicates a request<br> with a cookie
+    C->>+G: Request Web Page<br>[cookie]
+    G->>G: Consistent lookup of<br>server using cookie value
+    G->>+B: Request<br>[cookie]
+    B-->>-G: Response
+    G-->>-C: Response
+```
+
+#### Backend Initiated Sessions
+
+When dealing with backend-initiated sessions, the process becomes somewhat more complex. The gateway must modify the
+cookie's value to establish persistent sessions. However, in the most straightforward scenario, if the backend is not
```

Comment:
Or set additional cookies. It's a bag of cookies.

The gateway can't really remain passive - if you run a backend that requires persistence and gateway doesn't cooperate, it's broken. 

If we want to support 'backend initiated' sessions ( Servlet/PHP case) - gateway must abstain from generating a cookie, instead detect a cookie set by the backend. If yes - it can modify it or add its own second cookie to support the backend.